test/e2e-go/upgrades/application_support_test.go

```diff
@@ -448,18 +448,14 @@ int 1
 	a.NoError(err)
 	signedTxn, err = client.SignTransactionWithWallet(wh, nil, tx)
 	a.NoError(err)
-	_, err = client.BroadcastTransaction(signedTxn)
+	txid, err := client.BroadcastTransaction(signedTxn)
 	a.NoError(err)
 
-	curStatus, err = client.Status()
-	a.NoError(err)
-
-	round = curStatus.LastRound
-
-	client.WaitForRound(round + 2)
-	pendingTx, err = client.GetPendingTransactions(1)
+	// Try polling 10 rounds to ensure txn is committed.
+	round, err = client.CurrentRound()
 	a.NoError(err)
-	a.Equal(uint64(0), pendingTx.TotalTransactions)
```

Comment:
@algochoi For my understanding:  While investigating, did you understand why the assertion previously checking for 0 transactions?

Reply:
It looks like it's checking if the transaction was committed or not. If it was not committed due to network congestion or something, it would error, so this PR waits for a maximum of 10 rounds instead of two.

Reply:
Resolving - Had a quick verbal with @algochoi where I clarified a misunderstanding on my side.  

I mistakenly thought `GetPendingTransactions` retrieves recently confirmed transactions, which would have changed the test's intent.